Updating release documentation for clarity

willdavsmith opened the pull request:
# Description

* Updating release process documentation to make it more clear

## Type of change

<!--

Please select **one** of the following options that describes your change and delete the others. Clearly identifying the type of change you are making will help us review your PR faster, and is used in authoring release notes.

If you are making a bug fix or functionality change to Radius and do not have an associated issue link please create one now. 

-->

- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

<!--

Please update the following to link the associated issue. This is required for some kinds of changes (see above).

-->

Closes: #7106

vinayada1 commented on docs/contributing/contributing-releases/README.md:
```diff
@@ -1,115 +1,156 @@
+1. In the `radius-project/radius` repo, run the [Release verification](https://github.com/radius-project/radius/actions/workflows/release-verification.yaml) workflow. Run the workflow from the release branch (format: `release/x.y`) and use the Radius RC release version number being released.
 
-In the project-radius/samples repo, run the [Test Samples](https://github.com/project-radius/samples/actions/workflows/test.yaml) workflow. Run the workflow from the edge branch and using the Radius release or pre-release version number being released.
+1. In the `radius-project/samples` repo, run the [Test Samples](https://github.com/radius-project/samples/actions/workflows/test.yaml) workflow. Run the workflow from the `edge` branch and using the Radius release or pre-release version number being released.
```
fix numbering

willdavsmith replied:
I put all 1s here because markdown will do auto-numbering: https://github.com/radius-project/radius/blob/19715eea91bd1559078d1a8d496c40d977a556c9/docs/contributing/contributing-releases/README.md#creating-an-rc-release

vinayada1 commented on docs/contributing/contributing-releases/README.md:
```diff
@@ -1,115 +1,156 @@
-### Test tutorials and samples
+1. You may need to wait around ~20 minutes for the release assets to be built and published.
+
+1. There should be a GitHub workflow run in progress [here](https://github.com/radius-project/radius/actions/workflows/build.yaml) that was triggered by the `vx.y.z-rc1` tag. Monitor this workflow to ensure that it completes successfully. If it does, then the release candidate has been created.
```
fix numbering

willdavsmith replied:
same comment as above

willdavsmith commented on docs/contributing/contributing-releases/README.md:
````diff
@@ -1,115 +1,156 @@
+   ```
 
-1. Create a PR to merge into the release branch (format: release/x.y) in the radius repo. Cherry-pick the `version.yaml` changes and the release notes from the previous step in this PR. This will ensure that the version changes and release notes are included in the release branch. [Example](https://github.com/radius-project/radius/pull/6114/files)
+1. In your local branch, update the `version.json` file.
````
I think this one should be version.json because it is in the bicep repo

vinayada1 commented on .github/workflows/release.yaml:
```diff
@@ -37,7 +37,7 @@ jobs:
     name: Generate release note from template
     runs-on: ubuntu-latest
-    if: github.repository == 'radius-project/radius' && github.event_name == 'pull_request' && github.event.pull_request.head.ref == 'main'
+    if: github.repository == 'radius-project/radius' && github.event_name == 'pull_request' && github.base_ref == 'main'
```
Could we add comments in this workflow to explain the intended purpose?

kachawla commented on docs/contributing/contributing-releases/README.md:
```diff
@@ -1,115 +1,156 @@
+1. Verify that a release was created on Github Releases for the current version ([Example](https://github.com/radius-project/radius/releases)).
 
-1. Go through steps 1-4 of "Creating an RC release" above on the `main` branch, substituting the patch release version instead of the final release version. For example, if the final release version number is 0.1.0, the patch release version would be 0.1.1.
+1. In the `radius-project/radius` repo, run the [Release verification](https://github.com/radius-project/radius/actions/workflows/release-verification.yaml) workflow. Run the workflow from the release branch (format: `release/x.y`) and use the Radius RC release version number being released.
```
```suggestion
1. In the `radius-project/radius` repo, run the [Release verification](https://github.com/radius-project/radius/actions/workflows/release-verification.yaml) workflow. Run the workflow from the release branch (format: `release/x.y`) and use the Radius release version number being released.
```

kachawla commented on docs/contributing/contributing-releases/README.md:
```diff
@@ -1,115 +1,156 @@
+1. In the `radius-project/radius` repo, run the [Release verification](https://github.com/radius-project/radius/actions/workflows/release-verification.yaml) workflow. Run the workflow from the release branch (format: `release/x.y`) and use the Radius RC release version number being released.
 
-1. After creating the pull request, there should be an automatically generated release notes comment. Create a new release note document in the [release-notes](../../release-notes/) directory. Follow the directory's README.md for instructions on how to create a new patch release note document. Include this file in the release version pull request. [Example](https://github.com/project-radius/radiu ...
+1. In the `radius-project/samples` repo, run the [Test Samples](https://github.com/radius-project/samples/actions/workflows/test.yaml) workflow. Run the workflow from the `edge` branch and using the Radius release or pre-release version number being released.
```
```suggestion
1. In the `radius-project/samples` repo, run the [Test Samples](https://github.com/radius-project/samples/actions/workflows/test.yaml) workflow. Run the workflow from the `edge` branch and using the Radius release version number being released.
```

kachawla reviewed (commented):
Could you update the PR title and description to reflect a summary/description of the bug being fixed? Using 'bug from 0.30 release' is too generic and doesn't provide a clear commit message.

willdavsmith commented on docs/contributing/contributing-releases/README.md:
```diff
@@ -1,115 +1,156 @@
+1. In the `radius-project/radius` repo, run the [Release verification](https://github.com/radius-project/radius/actions/workflows/release-verification.yaml) workflow. Run the workflow from the release branch (format: `release/x.y`) and use the Radius RC release version number being released.
 
-In the project-radius/samples repo, run the [Test Samples](https://github.com/project-radius/samples/actions/workflows/test.yaml) workflow. Run the workflow from the edge branch and using the Radius release or pre-release version number being released.
+1. In the `radius-project/samples` repo, run the [Test Samples](https://github.com/radius-project/samples/actions/workflows/test.yaml) workflow. Run the workflow from the `edge` branch and using the Radius release or pre-release version number being released.
```
the release branch in samples and docs branches don't exist until we do the final release, so we should run against edge until we're ready to create them.

rynowak commented on docs/contributing/contributing-releases/README.md:
````diff
@@ -1,116 +1,158 @@
-1. Create a PR to merge into the release branch (format: release/x.y) in the radius repo. Cherry-pick the `version.yaml` changes and the release notes from the previous step in this PR. This will ensure that the version changes and release notes are included in the release branch. [Example](https://github.com/radius-project/radius/pull/6114/files)
+1. Clone the [radius-project/radius](https://github.com/radius-project/radius) repo locally, or use your existing local copy.
+   ```
+   git clone git@github.com:radius-project/radius.git
````
Should this be a fork rather than the main repo? Are we counting on the release captain to have push permissions?

willdavsmith replied:
Yes, for now we are counting on the release captain to have push permissions at the moment. This is because we require the GH token to create a comment on the PR. We could definitely change this but for now we should fork directly from the repo.

ref: https://github.com/radius-project/radius/issues/6927#issuecomment-1944388891

rynowak commented on docs/contributing/contributing-releases/README.md:
````diff
@@ -1,116 +1,158 @@
-1. Go through steps 1-4 of "Creating an RC release" above on the `main` branch, substituting the patch release version instead of the final release version. For example, if the final release version number is 0.1.0, the patch release version would be 0.1.1.
+1. Clone the [radius-project/bicep](https://github.com/radius-project/bicep) repo locally, or use your existing local copy.
+   ```
+   git clone git@github.com:radius-project/radius.git
````
Same question, should this be a fork?

rynowak reviewed (dismissed):
:shipit: This is a really big improvement. I had one question about whether the instructions should be talking about forks, or about the main repo. Other than that LGTM

kachawla reviewed (commented):
@willdavsmith is this pending on addressing comments or ready for review?

willdavsmith commented:
> @willdavsmith is this pending on addressing comments or ready for review?

ready for review! It was reviewed by @youngbupark during the last release and seems to be correct.

willdavsmith committed:
PR

Signed-off-by: willdavsmith <willdavsmith@gmail.com>

kachawla committed:
Merge branch 'main' into willdavsmith/0.30-release-fixes

kachawla commented:
> > @willdavsmith is this pending on addressing comments or ready for review?
> 
> ready for review! It was reviewed by @youngbupark during the last release and seems to be correct.

great, thanks! I'll merge it once all the workflows pass.

kachawla committed:
Merge branch 'main' into willdavsmith/0.30-release-fixes
